projects/prison-case-notes-to-probation/src/main/kotlin/uk/gov/justice/digital/hmpps/datetime/ZonedDateTimeDeserializer.kt

```diff
@@ -15,7 +15,7 @@ class ZonedDateTimeDeserializer : JsonDeserializer<ZonedDateTime>() {
     @Throws(IOException::class, JsonProcessingException::class)
     override fun deserialize(parser: JsonParser, context: DeserializationContext?): ZonedDateTime {
         val json = parser.text
-        return if (json.matches(".*(\\+|-)\\d{2}(:?\\d{2})?$".toRegex())) {
+        return if (json.matches(".*(?:Z|[+-](?:2[0-3]|[01]\\d):[0-5]\\d)(?:\\[.*\\])?".toRegex())) {
```

Comment:
Also, does this break for lowercase 'z' ?

Reply:
I've had a play around with the `parseDefaulting` method, which would allow us to define a date/time formatter with optional timezone/offset.  Here's an example, with it defaulting to UTC if no zone is provided: https://pl.kotl.in/fFh168dei.  Would that be better?  

I'm just a bit wary of using regex for this kind of thing, as it usually ends up being more error-prone and less readable.